ironfish/src/indexers/minedBlocksIndexer.ts

```diff
@@ -166,14 +157,7 @@ export class MinedBlocksIndexer {
     })
 
     this.accounts.onAccountRemoved.on(async (account) => {
-      await this.database.transaction(async (tx) => {
-        const hashes = await this.accountToBlockHashes.get(account.name, tx)
-        if (hashes) {
-          for (const hash of hashes) {
-            await this.minedBlocks.del(hash, tx)
-          }
-        }
-      })
+      await this.removeMinedBlocks(account.name)
```

Comment:
I think all this should be doing is queueing the "job" for removal, and letting the "scheduler" pick up that job. It ensures that no two things are running that could be conflicting. Like a prune job and updating the index.